src/dev/button.cpp

```diff
@@ -0,0 +1,25 @@
+#include <EVT/dev/button.hpp>
+#include <HALf3/stm32f3xx_hal.h>
+
+namespace EVT::core::DEV {
+Button::Button(IO::GPIO& gpio) : gpio(gpio) {
+    this->gpio = gpio;
+    this->timeSinceLastPress = 0;
+}
+
+IO::GPIO::State Button::getState() {
+    return this->gpio.readPin();
+}
+
+bool Button::debounce(int debounceTime) {
+    if (HAL_GetTick() - this->timeSinceLastPress > debounceTime) {
```

Comment:
This is ok for now, but this implementation is specific to the STM (`HAL_GetTick` won't exist in other platforms). We have a task in Trello for a platform independent `GetTick` equivalent. For now we can accept this

Reply:
Ok